### DISCUSSION TITLE
How can I disable automatic deploys ONLY for master

### DISCUSSION THREAD

**Discussion by kgoedecke** [vercel/vercel#5238] [Help] [answered]
I want to disable automatic deploys on Vercel, but only for the `master` branch. Vercel should still deploy automatically for `develop` and all other feature branches.

For master/our production system I want deploy manually with `now -A now.prod.json --prod`

Is there a way to achieve this?

**Comment by damianobarbati** [comment]
@kgoedecke  did you find a solution?

I have the same problem because I deploy production only for **tags** made on master branch

**Comment by yuchiko** [comment]
I've already wrote 3 letters to support. And no luck

**Comment by pcjmfranken** [comment]
Could always set up a DIY deployment system and even automate it through GH actions. Here are some ways you can control your deployment URL: https://github.com/vercel/community/discussions/114#discussioncomment-1897657

**Comment by barelyhuman** [comment]
Project > Settings > Git > Scroll down to __Ignored Build Step__

Change it to match the following 
```sh
if [ $(git rev-parse --abbrev-ref HEAD) != "master" ]; then git diff --quite HEAD^ HEAD ./ ; else exit 0; fi
```

### Explanation 
Check if the branch is not `master`, 
then run the normal hash diffing to trigger deploys 
if it's master then just exit with 0 to avoid creating a build 

For more info:
https://vercel.com/docs/concepts/projects/overview#ignored-build-step

**Reply by tahaygun** [reply]
For me, somehow all branches are behaving like master. I learned about it while printing the branch name. Interesting.

**Reply by barelyhuman** [reply]
I’m not sure that should be happening. 

a screenshot of that print or if it’s a public project, you can share the settings of it so 
@leerob and team can validate

**Reply by tahaygun** [reply]
Sure, this is from the latest version, I thought maybe deriving the branch name in another way would work but still the same result as 'git rev-parse --abbrev-ref HEAD' command. Btw, I want to make sure that the preview is built only for master.
![image]

**Reply by amyegan** [reply, marked as answer]
A relatively new [`git.deploymentEnabled`](https://vercel.com/docs/concepts/projects/project-configuration/git-configuration#git.deploymentenabled) configuration option is available to let you disable automatic deployment to specified branches or an entire repo.

**Reply by tahaygun** [reply]
sounds awesome! thanks for the update.
